### PR TITLE
Decapitate Decapitated

### PR DESCRIPTION
**Pull request by Surrealaser**
-No more RNG chance of decapitation from damage if wearing a helmet, and not a synth.

### CODE REVIEW

File: code/modules/organs/limbs.dm [hunk 1]
```diff
@@ -225,7 +225,9 @@
 	src.update_damages()
 
 	//If limb took enough damage, try to cut or tear it off
-	if(body_part != UPPER_TORSO && body_part != LOWER_TORSO && !no_limb_loss)
+	var/obj/item/clothing/worn_helmet = owner.head
+	if(body_part != UPPER_TORSO && body_part != LOWER_TORSO && !no_limb_loss && (body_part != HEAD && !istype(worn_helmet, /obj/item/clothing/head/helmet) && !(owner.species.flags & IS_SYNTHETIC))) //Synth heads can come off, not human ones for gameplay/balance reasons... if they're wearing a helmet.
```
Comment (Rohesie) [inline code/modules/organs/limbs.dm]: Current logic doesn't work, try an early return and a more clear check.

**Comment by Ghommie** [comment]
I'd say it's preferable to check for the headwear melee protection rather than the path's, and the decap chance lowering as the stat raises, with the 0% point at about the standard marine's helmet stat.

**Comment by Surrealaser** [comment]
> It's preferable to check for the headwear melee protection rather than the path's, so to include all sufficiently armored headwear, not just types.
> Optionally, have the decap chance lowering as the stat raises, with the 0% point at about the standard marine's helmet stat.

To be clear, I am prioritizing balance and fun over realism here, hence using only the helmet path.

**Comment by Rohesie** [comment]
Ghommie's suggestion preserves balance, with helmets providing 100% defense against decaps.

**Comment by Surrealaser** [comment]
> Ghommie's suggestion preserves balance, with helmets providing 100% defense against decaps.

Put it this way: I'm not sure why the added complexity is needed for an ultra niche category of helmets that basically never see use.

File: code/modules/organs/limbs.dm [hunk 2]
```diff
@@ -225,12 +225,17 @@
...
 	//If limb took enough damage, try to cut or tear it off
-	if(body_part != UPPER_TORSO && body_part != LOWER_TORSO && !no_limb_loss)
-		if(config.limbs_can_break && brute_dam >= max_damage * config.organ_health_multiplier)
-			var/cut_prob = brute/max_damage * 10
-			if(prob(cut_prob))
-				droplimb()
-				return
+
+	if(body_part == UPPER_TORSO || body_part == LOWER_TORSO || no_limb_loss)
+		return
```
Comment (Rohesie) [inline code/modules/organs/limbs.dm]: Update health before you return.

File: code/modules/organs/limbs.dm [hunk 3]
```diff
@@ -225,12 +225,17 @@
...
+	var/obj/item/clothing/worn_helmet = owner.head
+	if(body_part == HEAD && istype(worn_helmet, /obj/item/clothing/head/helmet) && !(owner.species.flags & IS_SYNTHETIC) ) //Early return if the body part is a head but target is wearing a helmet and is not a synth
+		return
```
Comment (Rohesie) [inline code/modules/organs/limbs.dm]: Here as well.

**Pull request merged**